modules/internal/ChapelRange.chpl

```diff
@@ -2381,8 +2384,8 @@ proc _cast(type t, r: range(?)) where isRangeType(t) {
           var a: intIdxType;
           f <~> a;
           _alignment = a;
-        } else {
-          halt("Trying to read an aligned range value into a non-stridable array");
+        } else if boundsChecking {
+          HaltWrappers.boundsCheckHalt("Trying to read an aligned range value into a non-stridable array");
```

Comment:
This doesn't quite seem like bounds checking, but I'm not sure what I'd call it instead